tensorflow/lite/micro/tools/benchmarking/collect_meta_data.sh

```diff
@@ -0,0 +1,159 @@
+
+
+set -e
+
+source ${TENSORFLOW_ROOT}tensorflow/lite/micro/tools/ci_build/helper_functions.sh
+
+function substitute_strings() {
+  search="// %%%_$1_%%%"
+  lines=$(fold -w 90 -s <<< "$2")
+  SAVED_IFS=${IFS}
+  IFS=$'\n' lines_array=( ${lines} )
+  IFS=${SAVED_IFS}
+  replacement=()
+  for line in "${lines_array[@]}"; do
+    line=$(sed -e 's/"/\\"/g' <<< "${line}")
+    line=$(printf '"%s",\n    ' "${line}")
+    replacement+=( "${line}" )
+  done
+
+  tempfile=$(mktemp)
+
+  SEARCH_PATTERN="$search" REPLACEMENT_PATTERN="${replacement[@]}" awk '
+    BEGIN {
+        search = ENVIRON["SEARCH_PATTERN"]
+        replacement = ENVIRON["REPLACEMENT_PATTERN"]
+    }
+    s = index($0,search) {
+        $0 = substr($0,1,s-1) replacement substr($0,s+length(search))
+    }
+    { print }
+  ' "${GENERATED_FILE}" > ${tempfile}
+  mv ${tempfile} "${GENERATED_FILE}"
+}
+
+mkdir -p $(dirname ${GENERATED_FILE})
+cp -p ${TEMPLATE_FILE} ${GENERATED_FILE}
+
+# model analysis and SHA1
+if [[ ${MODEL_FILE} ]]; then
+  result=$(python3 \
+    "${TENSORFLOW_ROOT}tensorflow/lite/micro/tools/benchmarking/analyze_model.py" \
+    --model_file="${MODEL_FILE}" \
+    )
+  substitute_strings model_analysis_strings "${result}"
+
+  result=$(shasum -b "${MODEL_FILE}" | cut -f 1 -d ' ')
+  substitute_strings model_sha1_strings "${result}"
+fi
+
+# compile date
+result=$(date)
+substitute_strings compilation_date_strings "${result}"
+
+GIT_TENSORFLOW_ROOT="${TENSORFLOW_ROOT:-./}"
+set +e
+# Git repo commit information
+result=$(cd ${GIT_TENSORFLOW_ROOT} && git rev-parse --verify HEAD)
+if [[ $? != 0 ]]; then
+  result="<git commit information not available>"
+fi
+substitute_strings git_commit_strings "${result}"
+
+# Git repo status information
+result=$(cd ${GIT_TENSORFLOW_ROOT} && git status)
+if [[ $? != 0 ]]; then
+  result="<git status information not available>"
+fi
+substitute_strings git_status_strings "${result}"
+set -e
+
+# Compiler information
+result="${CC}"
+substitute_strings cc_name_strings "${result}"
+result=$("${CC}" --version)
+substitute_strings cc_version_strings "${result}"
+result="${CC_FLAGS}"
+substitute_strings cc_flags_strings "${result}"
+
+result="${CXX}"
+substitute_strings cxx_name_strings "${result}"
+result=$("${CXX}" --version)
+substitute_strings cxx_version_strings "${result}"
+result="${CXX_FLAGS}"
+substitute_strings cxx_flags_strings "${result}"
+
+result="kernel= ${KERNEL_OPTIMIZATION}"
+result+="  core= ${CORE_OPTIMIZATION}"
+result+="  third-party-kernel= ${THIRD_PARTY_KERNEL_OPTIMIZATION}"
+substitute_strings optimization_flag_strings "${result}"
+
+# Target information
+TARGET="${TARGET:-linux}"
+TARGET_ARCH="${TARGET_ARCH:-x86}"
+OPTIMIZED_KERNEL="${OPTIMIZED_KERNEL:-none}"
+BUILD_TYPE="${BUILD_TYPE:-default}"
+result=$(printf 'TARGET=%s\nTARGET_ARCH=%s\nOPTIMIZATION=%s\nBUILD_TYPE=%s\n' \
+  "${TARGET}" \
+  "${TARGET_ARCH}" \
+  "${OPTIMIZED_KERNEL}" \
+  "${BUILD_TYPE}" \
+)
+if [[ "${XTENSA_CORE}" ]]; then
```

Comment:
We should add more of the XTENSA metadata defined [here](https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/tools/make/targets/xtensa_makefile.inc#L17).

Specifically, XTENSA_BASE and XTENSA_TOOLS_VERSION would be super useful. It'd probably also be good to figure add in the nnlib url/md5, but that's definitely more complicated than the CMSIS one.

Reply:
done